start.R

```diff
@@ -244,60 +236,78 @@ if (any(c("--reprepare", "--restart") %in% flags)) {
-     if ("--reprepare" %in% flags) c("full.gms" = "full_beforeRestart.gms",
-                                     "fulldata.gdx" = "fulldata_beforeRestart.gdx")
-                  )
-  message("\n", paste(names(filestomove), collapse = ", "), " will be moved and get a postfix '_beforeRestart'.\n")
-  if(! exists("slurmConfig")) slurmConfig <- choose_slurmConfig()
-  if ("--quick" %in% flags) slurmConfig <- paste(slurmConfig, "--time=60")
-  message()
-  for (outputdir in outputdirs) {
-    message("Restarting ", outputdir)
-    load(paste0("output/", outputdir, "/config.Rdata")) # read config.Rdata from results folder
-    cfg$restart_subsequent_runs <- restart_subsequent_runs
-    # for debug, testOneRegi, quick: save original settings to cfg$backup; restore them from there if not set.
-    if ("--debug" %in% flags) {
-      if (is.null(cfg[["backup"]][["cm_nash_mode"]])) cfg$backup$cm_nash_mode <- cfg$gms$cm_nash_mode
-      cfg$gms$cm_nash_mode <- "debug"
-    } else {
-      if (! is.null(cfg[["backup"]][["cm_nash_mode"]])) cfg$gms$cm_nash_mode <- cfg$backup$cm_nash_mode
-    }
-    cfg$gms$cm_quick_mode <- if ("--quick" %in% flags) "on" else "off"
-    if (any(c("--quick", "--testOneRegi") %in% flags)) {
-      if (is.null(cfg[["backup"]][["optimization"]])) cfg$backup$optimization <- cfg$gms$optimization
-      cfg$gms$optimization <- "testOneRegi"
-      if (testOneRegi_region != "") cfg$gms$c_testOneRegi_region <- testOneRegi_region
-    } else {
-      if (! is.null(cfg[["backup"]][["optimization"]])) cfg$gms$optimization <- cfg$backup$optimization
-    }
-    if ("--quick" %in% flags) {
-      if (is.null(cfg[["backup"]][["cm_iteration_max"]])) cfg$backup$cm_iteration_max <- cfg$gms$cm_iteration_max
-      cfg$gms$cm_iteration_max <- 1
-    } else {
-      if (! is.null(cfg[["backup"]][["cm_iteration_max"]])) cfg$gms$cm_iteration_max <- cfg$backup$cm_iteration_max
-    }
-    if (! "--test" %in% flags) {
-      filestomove_exists <- file.exists(file.path("output", outputdir, names(filestomove)))
-      file.rename(file.path("output", outputdir, names(filestomove[filestomove_exists])),
-                  file.path("output", outputdir, filestomove[filestomove_exists]))
+  if ("--gamscompile" %in% flags) {
+    for (outputdir in outputdirs) {
+      load(file.path("output", outputdir, "config.Rdata")) # read config.Rdata from results folder
+      tmpModelFile <- sub(".gms", paste0("_", cfg$title, ".gms"), cfg$model)
+      if (file.exists(file.path("output", outputdir, cfg$model))) {
+        file.copy(file.path("output", outputdir, cfg$model), tmpModelFile)
+        manipulateConfig(tmpModelFile, cfg$gms)
+        command <- paste("gams", tmpModelFile, "-a=c -errmsg=1 -pw=$( tput cols ) -ps=0",
+                         "&& rm $( echo ", tmpModelFile, " | sed 's/\\.[^\\.]\\+/.lst/' ) || less -j 4",
+                         "--pattern='^\\*\\*\\*\\*' $( echo", tmpModelFile, "| sed 's/\\.[^\\.]\\+/.lst/' )")
```

Comment:
Maybe it doesn't work on Windows at all, the code comes from `/p/projects/rd3mod/tools/gamscompile`. If it works only on the cluster anyway, that I can call `gamscompile` directly similar to `nashstat` [in prepare_and_run.R](https://github.com/remindmodel/remind/blob/32c84448bbba381c286f2d429b913c53f51637aa/scripts/start/prepare_and_run.R#L1116-L1128)

Reply:
What I think this does is:
 
- runs gams with only the compile option (1st line)
- if no compile errors, removes listings file (2nd line)
- if there are compile errors, open the listings file in `less` file viewer and jump to the first line starting with `****`. (2nd/3rd line)

the `$( tput cols )` (giving the terminal width) and everything from line 2 looks Linux-specific.  
I would carefully opt for removing line 2 and 3, since they are not really part of the compilation process? With the side benefit of Windows compatibility (if we also replace `$( tput cols )` with a number)

Reply:
I cleaned the whole code. The files stay in `./output/gamscompile` so you can have a look. If you run `--gamscompile` on a run with the same title, it is overwritten, but I think this is rather a boon, not a bug.